### PR TITLE
Modify db package to support python3 (DM-7253)

### PR DESCRIPTION
**Pull request by andy-slac**

### CODE REVIEW

File: python/lsst/db/engineFactory.py [hunk 1]
```diff
@@ -25,9 +25,11 @@
 
 @author  Jacek Becla, SLAC
 """
+from future import standard_library
+standard_library.install_aliases()
 
 # standard library imports
-from ConfigParser import ConfigParser, NoSectionError
+from configparser import ConfigParser, NoSectionError
```
Comment (timj) [inline python/lsst/db/engineFactory.py]: I see that future versions of future are dropping `configparser` and recommending that people install the PyPI compatibility version on python 2. The alternative is to try/except:
```python
try:
   from ConfigParser import ...
except ImportError:
   from configparser import ...
```

File: python/lsst/db/testHelper.py [hunk 2]
```diff
@@ -71,10 +77,11 @@ class CannotExecuteScriptError(InvalidRequestError):
 # Note: 'read_default_group' is not supported, since it can cause
 # the MySQLdb driver and the mysql executable to connect differently.
 optionToConnectArgMap = \
-    dict((v, k) for (k, v) in connectArgToOptionMap.iteritems())
+    dict((v, k) for (k, v) in connectArgToOptionMap.items())
```
Comment (timj) [inline python/lsst/db/testHelper.py]: Please change to a dict comprehension.

Comment (andy-slac) [inline python/lsst/db/testHelper.py]: OK, will do

File: python/lsst/db/engineFactory.py (hunk 1 above)
Comment (andy-slac) [inline python/lsst/db/engineFactory.py]: I'll change that to try ... except

**Pull request merged**